cupboard_app/queries.py

```diff
@@ -267,3 +268,88 @@ def create_list_ingredient(ingredient: str, amount: int | float, unit: str) -> d
         }
 
     return ingredient_dict
+
+def get_user_lists_ingredients(username: str, id: int = None) -> QuerySet:
+    """
+    Gets all lists for the specific user from the database.
+
+    Args:
+        username: User's username
+        id: User ID
+
+    Returns:
+        QuerySet of all the lists for the specific user.
+    """
+    if id:
+        result = UserListIngredients.objects.filter(
+            user__id=id
+        )
+    else:
+        result = UserListIngredients.objects.filter(
+            user__username=username
+        )
+
+    return result
+
+def update_list_ingredient(
+    username: str,
+    listName: str,
+    ingredient: str,
+    amount: int | float,
+    unit: str
+) -> str:
+    """
+    Updates an ingredient in the user's list.
+
+    If ingredient exists in the user's list, then update the given ingredient's
+    amount and unit in the passed amount and unit.
+
+    If ingredient does not exist in the user's list, then add new ingredient
+    dictionary to user's list.
+
+    Args:
+        username: User's username
+        listName: Name of the list
+        ingredient: Name of the ingredient
+        amount: Quantity of the ingredient
+        unit: The unit of measure for the ingredient
+    """
+    result = UPDATE_FAILED_MSG
+    try:
+        list_ingredient = create_list_ingredient(
+            ingredient=ingredient,
+            amount=amount,
+            unit=unit
+        )
+        if list_ingredient:
+            # Check list exists
+            user_list = UserListIngredients.objects.filter(
+                user__username=username,
+                listName__listName=listName
+            ).first()
+            if user_list:
+                search_id = list_ingredient.get('ingredientId')
+                unit_id = list_ingredient.get('unitId')
+                # Check if ingredient exists
+                if not any(
```

Comment:
I think the team was talking about how we wanted to update the ingredient if it was the same ingredient and unit, otherwise add a new ingredient.

Please check with others on this.